web/src/message_list_view.js

```diff
@@ -780,6 +787,17 @@ export class MessageListView {
     _get_message_template(message_container) {
         const msg_reactions = reactions.get_message_reactions(message_container.msg);
         message_container.msg.message_reactions = msg_reactions;
+
+        // Here we set up some properties of message_container, that shall help us to
+        // determine where to display the typing notification for message editing when
+        // only a single message is rendered.
+        const include_sender = message_container.include_sender;
+        const is_hidden = message_container.is_hidden;
+        const status_message = Boolean(message_container.status_message);
+        message_container.editing_in_left_col = !include_sender && !is_hidden;
+        message_container.editing_alongside_sender = include_sender && !status_message;
+        message_container.editing_status_msg = include_sender && status_message;
```

Comment:
When a new message is sent to a stream and is edited such that edit happens in the left_col , then for the clients the typing notification for editing won't appear. Hence, this section is added when only a single message is rendered.

Reply:
I'm pretty sure the block in https://github.com/zulip/zulip/pull/28585/files#r1453938697 is expected to run every time a message is being rendered; and it looks like it only inspects immutable properties of the message, so I'd expect it to compute the same value every time. I would try deleting this and then debugging with `console.log` or breakpoints or something why that isn't working for you. From an abstractions standpoint, the other code path is where this belongs.